### ISSUE TITLE
Promos: Heading and Anchor nesting?

### ISSUE THREAD

**Issue by micmath** [bbc/gef-docs#33]
Our guidance is currently for this structure:
```html
<h4>
  <a href="/to/permalink/1">Headline text goes here</a>
</h4>
```
Would the following variation be acceptable?
```html
<a href="/to/permalink/1"><h4>Headline text goes here</h4></a>
```

**Comment by greenc05** [comment]
The variation of link with nested heading is needed sometimes, if for example if adding a label such as 'Breaking news' or 'Live', this is not necessarily part of the heading but it is part of the link text.

**Comment by greenc05** [comment]
Some testing shows that VoiceOver Mac/Safari does not announce heading semantics if we have (note all other screen readers worked well when the h3 had a nested link) 

```
<a href="my-link">
    <h3>My heading</h3>
</a>
```

**Comment by micmath** [comment]
Is there a spec defined somewhere (don't laugh) for how a standard screen reader should treat that scenario? i assume it is standard, valid and well-formed HTML.

**Comment by Heydon** [comment]
I don't think so. But if there is, you can bet Apple/Voiceover won't follow it...

**Comment by micmath** [comment]
🤦I'm reminded of the browser wars back in the 00's. We lost a lot of good soldiers...

**Comment by Heydon** [comment]
![The Browser Wars: 1995 to 2019](https://user-images.githubusercontent.com/683558/57444794-dea3cf80-7248-11e9-8d6e-3fda12e24b52.png)

**Comment by EmmaJP** [comment]
As a link element is consider both flow content and phrasing content, I think it can legitimately contain other flow elements.
https://www.w3.org/TR/html52/document-metadata.html#the-link-element

**Comment by Heydon** [comment]
Yeah, HTML5 permits 'block links' but screen readers have diverging ways of handling them.

**Comment by micmath** [comment]
@EmmaJP I only ask about standards because if we are following the Rules of HTML and Apple/Voiceover are doing the wrong thing (as suggested by the fact that other AT combinations get it right) then that makes me think we _could_ leave it to Apple to solve the problem, rather than trying to fix their mistake in our code.

Of course, as we battle-hardened browser war survivors all know, being right counts for nothing when a web-client does it wrong and also has 60% of the market share.

**Comment by EmmaJP** [comment]
@micmath I'm in agreement with you. I believe our line is to follow the spec, as that is what others should also be doing. We can't cater for all the nuances and possible ways someone might do it differently. 

We did the same thing back in the browser war days, and survived longer than IE5.5 or 6.
